backend/src/test/java/fk/prof/backend/AssociationApiTest.java

```diff
@@ -219,6 +222,71 @@ public void getAssociationProxiedToLeader(TestContext context) throws Interrupte
     });
   }
 
+  @Test(timeout = 20000)
+  public void getAllAssociationsProxiedToLeader(TestContext context) throws InterruptedException {
+    final Async async = context.async();
+
+    VerticleDeployer leaderHttpDeployer = new LeaderHttpVerticleDeployer(vertx, config, backendAssociationStore, policyStore);
+    Runnable leaderElectedTask = LeaderElectedTask.newBuilder().build(vertx, leaderHttpDeployer, backendAssociationStore, policyStore);
+
+    VerticleDeployer leaderParticipatorDeployer = new LeaderElectionParticipatorVerticleDeployer(vertx, config, curatorClient, leaderElectedTask);
+    VerticleDeployer leaderWatcherDeployer = new LeaderElectionWatcherVerticleDeployer(vertx, config, curatorClient, inMemoryLeaderStore);
+    CompositeFuture.all(leaderParticipatorDeployer.deploy(), leaderWatcherDeployer.deploy()).setHandler(deployResult -> {
+      if(deployResult.failed()) {
+        context.fail(deployResult.cause());
+      }
+
+      try {
+        //This sleep should be enough for leader store to get updated with the new leader and leader elected task to be executed
+        Thread.sleep(5000);
+        when(inMemoryLeaderStore.isLeader()).thenReturn(false);
+
+        Future<ProfHttpClient.ResponseWithStatusTuple> r1 = makeRequestReportLoad(BackendDTO.LoadReportRequest.newBuilder().setIp("1").setPort(1).setLoad(0.5f).setCurrTick(2).build());
+        Future<ProfHttpClient.ResponseWithStatusTuple> r2 = makeRequestReportLoad(BackendDTO.LoadReportRequest.newBuilder().setIp("2").setPort(1).setLoad(0.5f).setCurrTick(2).build());
+        CompositeFuture.all(r1, r2).setHandler(ar -> {
+          if(ar.failed()) {
+            context.fail(ar.cause());
+          }
+          try {
+            Recorder.ProcessGroup.Builder pgBuilder = Recorder.ProcessGroup.newBuilder().setAppId("a").setCluster("c");
+            Recorder.ProcessGroup pg1 = pgBuilder.clone().setProcName("p1").build();
```

Comment:
I am using the same builder for creating multiple process groups. Protobuf builder objects cannot be used to create a new message instance if build is called on them already